recipes/nmslib/all/conanfile.py

```diff
@@ -1,66 +1,81 @@
+    }
 
-    _cmake = None
+    def export_sources(self):
+        export_conandata_patches(self)
 
-    def validate(self):
-        if self.settings.compiler == "Visual Studio":
-            if self.settings.compiler.version == "14":
-                raise ConanInvalidConfiguration("Visual Studio 14 builds are not supported")  # TODO: add reason in message
-            if self.options.shared:
-                raise ConanInvalidConfiguration("Visual Studio shared builds are not supported (.lib artifacts missing)")
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
+            self.options.rm_safe("fPIC")
 
-    def config_options(self):
-        if self.settings.os == "Windows":
-            del self.options.fPIC
+    def layout(self):
+        cmake_layout(self, src_folder="src")
 
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
+    def validate(self):
+        check_min_vs(self, 190)  # TODO: add reason in message
+        if is_msvc(self) and self.options.shared:
+            raise ConanInvalidConfiguration(
+                "Visual Studio shared builds are not supported (.lib artifacts missing)"
+            )
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version], strip_root=True,
-                  destination=self._source_subfolder)
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
 
-    def _configure_cmake(self):
-        if self._cmake is None:
-            cmake = CMake(self)
-            cmake.definitions["WITHOUT_TESTS"] = True
-            cmake.configure()
-            self._cmake = cmake
-        return self._cmake
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["WITHOUT_TESTS"] = True
+        tc.generate()
 
     def build(self):
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            tools.patch(**patch)
-        cmake = self._configure_cmake()
+        apply_conandata_patches(self)
+        cmake = CMake(self)
+        cmake.configure(build_script_folder=os.path.join(self.source_folder, "similarity_search"))
         cmake.build()
 
     def package(self):
-        self.copy(pattern="LICENSE", dst="licenses", src=self._source_subfolder)
-        cmake = self._configure_cmake()
+        copy(self, "LICENSE",
+             dst=os.path.join(self.package_folder, "licenses"),
+             src=self.source_folder)
+        cmake = CMake(self)
         cmake.install()
+        fix_apple_shared_install_name(self)
 
     def package_info(self):
         self.cpp_info.libs = ["NonMetricSpaceLib"]
-        if self.settings.os == "Linux":
-            self.cpp_info.system_libs = ["pthread"]
+        if self.settings.os in ["Linux", "FreeBSD"]:
+            self.cpp_info.system_libs = ["pthread", "m", "mvec"]
```

Comment:
```suggestion
            self.cpp_info.system_libs = ["pthread", "m"]
            if self.settings.arch in ["x86", "x86_64"]:
                self.cpp_info.system_libs.append("mvec")
```

The `mvec` library from gcc is only available on Intel systems as far as I know - I suspect this has come up before where users on aarch64 Linux had issues with `mvec` being unconditionally listed as a system lib

Reply:
I agree, it looked suspect to me as well. On my Ubuntu machine the ld script in `libm.a` adds `mvec` already adds transitively, so it is probably not necessary to separately list it in the recipe:
```
/* GNU ld script
*/
OUTPUT_FORMAT(elf64-x86-64)
GROUP ( /usr/lib/x86_64-linux-gnu/libm-2.31.a /usr/lib/x86_64-linux-gnu/libmvec.a )
`